ocaml/xe-cli/newcli.ml

```diff
@@ -300,10 +298,14 @@ let handle_unmarshal_failure ex ifd = match ex with
 let assert_upload_filename_in_args filename permitted_filenames =
   (* Prefix match instead of exact match is used here to workaround the old xva format that request files
    * are not in the command line *)
-  let is_filename_permited = List.exists (fun file_in_arg -> Astring.String.is_prefix ~affix:file_in_arg filename)
-      permitted_filenames in
+  let forbidden_path = ".." in
+  let is_filename_permited =  permitted_filenames
+                              |> List.exists (fun file_in_arg -> Astring.String.is_prefix ~affix:file_in_arg filename)
+                              && not (Stdext.Xstringext.String.has_substr filename forbidden_path) in
```

Comment:
only check the filename include the `..` may block the normal cases that working well before.
See the user input the path like `../a.file`, the server will ask the client to upload this file but it will fail with this change.
Maybe we can use canonicalized path or give some more specific check on the `..`.
However, we need to think about the effort to remove the old style of XVA, after this we can have the whole string name checking which is the best choice.

Reply:
updated